secant/UI Components/CheckCircle/CheckCircleStore.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  CheckCircleStore.swift
+//  secant-testnet
+//
+//  Created by Michal Fousek on 26.09.2022.
+//
+
+import Foundation
+import ComposableArchitecture
+import SwiftUI
+
+typealias CheckCircleReducer = Reducer<CheckCircleState, CheckCircleAction, CheckCircleEnvironment>
+typealias CheckCircleStore = Store<CheckCircleState, CheckCircleAction>
+typealias CheckCircleViewStore = ViewStore<CheckCircleState, CheckCircleAction>
+
+// MARK: - State
+
+struct CheckCircleState: Equatable {
+    var isChecked: Bool
+}
+
+// MARK: - Action
+
+enum CheckCircleAction: Equatable {
+    case updateIsChecked
+}
+
+// MARK: - Environment
+
+struct CheckCircleEnvironment { }
+
+extension CheckCircleEnvironment {
+    static let live = CheckCircleEnvironment()
+    static let mock = CheckCircleEnvironment()
+}
+
+// MARK: - Reducer
+
+extension CheckCircleReducer {
+    static let `default` = CheckCircleReducer { state, action, _ in
+        switch action {
+        case .updateIsChecked:
+            state.isChecked.toggle()
+            return .none
+        }
+    }
+}
+
+// MARK: - Store
+
+extension CheckCircleStore {
+    static func mock(isChecked: Bool) -> CheckCircleStore {
+        return CheckCircleStore(
+            initialState: CheckCircleState(isChecked: isChecked),
+            reducer: .default,
+            environment: CheckCircleEnvironment()
+        )
+    }
+}
+
+// MARK: - ViewStore
+
+extension CheckCircleViewStore {
+    static let placeholder = CheckCircleStore(
+        initialState: .checkedPlaceholder,
+        reducer: .default,
+        environment: CheckCircleEnvironment()
```

Comment:
Since you already have .mock static `CheckCircleEnvironment` we should use it here and consider to use it also in `placeholder`. We should try to avoid unnecessary environments instantiations.